pkg/object/s3.go

```diff
@@ -396,6 +396,8 @@ func newS3(endpoint, accessKey, secretKey string) (ObjectStorage, error) {
 	}
 	if accessKey != "" {
 		awsConfig.Credentials = credentials.NewStaticCredentials(accessKey, secretKey, "")
+	} else if accessKey == "" && secretKey == "" {
+		awsConfig.Credentials = credentials.AnonymousCredentials
```

Comment:
When the accessKey is empty, we usually use IAM role or other temporary token to access S3.

What's the current behavior to access public buckets? And, what's the use case that a public bucket is used with JuiceFS?

Reply:
For public buckets. the juicefs sync can't downloads the file from s3. If the IAM is disabled and no temporary token , I can't use juicefs sync.

Reply:
I see. The current patch will break IAM, we need another way to support anonymous access

Reply:
ok, I will close the pr, and pull an issue.